modules/ecs-service/scripts/deploy-ecs-service.py

```diff
@@ -0,0 +1,375 @@
+
+    # Assert that the service exists in the cluster
+    if len(res["services"]) == 0:
+        raise Exception("The service does not exist in the cluster")
+    elif len(res["services"]) > 1:
+        raise Exception(
+            "More than one service was found in the cluster with the same name"
+        )
+    logger.info(f"Service {service} found in cluster {cluster}")
+
+    # Log the current task definition
+    current_task_definition = res["services"][0]["taskDefinition"]
+    logger.info(f"Active task definition: {current_task_definition}")
+
+    # Log the last deployment time
+    last_deployment_time = res["services"][0]["deployments"][0]["createdAt"]
+    latest_revision = res["services"][0]["deployments"][0]["taskDefinition"].split(":")[
+        -1
+    ]
+    logger.info(
+        f"Latest completed deployment: {last_deployment_time} to deploy revision {latest_revision}"
+    )
+
+    # Get the task definition details
+    res = client.describe_task_definition(taskDefinition=current_task_definition)
+
+    # Log the image name and tag
+    image = res["taskDefinition"]["containerDefinitions"][0]["image"]
+    logger.info(f"Current revision ({latest_revision}) image: {image}")
+
+    return res["taskDefinition"], image
+
+
+def assertServiceIsStable(
+    client: any, cluster: str, service: str, timeout: int
+) -> None:
+    """Assert that the service is in a stable state (not in the middle of a deployment).
+
+    Args:
+        client (any): the low level boto3 ECS client
+        cluster (str): the name of the ECS cluster
+        service (str): the name of the ECS service
+        timeout (int): the number of seconds to wait to update the service
+
+    Raises:
+        Exception: Timeout exceeded: service never reached a stable state to perform deployment...
+    """
+    starttime = time.time()
+    endtime = starttime + timeout
+    while True:
+        if time.time() > endtime:
+            raise Exception(
+                f"Timeout exceeded: service never reached a stable state to perform deployment..."
+            )
+
+        # Assert that the service is in a stable state (not in the middle of a deployment)
+        res = client.describe_services(cluster=cluster, services=[service])
+
+        # Creates a list of unique deployment statuses. Using set notation should result in a list of
+        # length 1 if the service is stable. If the service is not stable, the list will contain more
+        # than one unique deployment status.
+        deployment_statuses = list(
+            set(
+                [
+                    deployment["status"]
+                    for deployment in res["services"][0]["deployments"]
+                ]
+            )
+        )
+        if deployment_statuses == ["PRIMARY"]:
+            logger.info(
+                f"Service {service} is in a stable state after {round(time.time() - starttime)}s"
+            )
+            break
```

Comment:
It definitely won't infinite loop, it will at the least raise an exception for a timeout. But yes I manually tested this a few times. I was trying to use the script in the go tests but for whatever reason I wasn't able to get the os module to run executables on windows... because... windows.

Reply:
If you really want, once I get a mac that is not M1 I can add this to be used in the tests than only run that test if it is not windows.

Reply:
This is kinda what I wanted to do ^^